Documentation/storage-providers.md

```diff
@@ -0,0 +1,89 @@
+* Common build scripts for building the operator images are in the main
+  [Rook repo](https://github.com/rook/rook/tree/master/build)].
+* Each provider has its own repo under the [Rook org](https://github.com/rook).
+  * Multiple community members are given push access to the repo, including
+    owners of the storage provider, Rook steering committee members,
+    and other Rook maintainers if deemed helpful or necessary by the steering
+    committee. Maintainers for the new provider are added according to the
+    [governance](https://github.com/rook/rook/blob/master/GOVERNANCE.md).
+  * Providers added to Rook prior to 2020 are grandfathered into the main
+    [Rook repo](https://github.com/rook/rook).
+* Storage providers must follow the Rook [governance](https://github.com/rook/rook/blob/master/GOVERNANCE.md)
+  in the interest of the good of the overall project. Storage providers have
+  autonomy in their feature work, while collaboration
+  with the community is expected where there are shared features.
+* A quarterly release cadence is in place for the operators in the main Rook repo.
+  Operators in their own repo define their own cadence and versioning scheme as desired.
+  * Storage providers own their release process, while following Rook best practices to
+    ensure high quality.
+* Docker images are pushed to the [Rook DockerHub](https://hub.docker.com/u/rook) where
+  each storage provider has its own repo.
+* Documentation for the storage provider is to be written by the storage provider
+  members. The build process will publish the documentation to the [Rook website](https://rook.github.io/docs/rook/master/).
+* All storage providers are added to the [Rook.io website](https://rook.io/)
+* A great Slack community is available where you can rally your troops.
+
+## Considering Joining Rook?
+
+If you own a storage provider and are interested in joining the Rook project to create
+an operator, please consider the following:
+
+* You are making a clear commitment to the development of the storage provider.
+  Creating an operator is not a one-time engineering cost, but is a long term commitment
+  to the community.
+* Support for a storage provider in Rook requires dedication and community support.
+* Do you really need an operator? Many storage applications (e.g. CSI drivers)
+  can be deployed with tools such as a Helm chart and don't really need the
+  flexibility of an operator.
+* Will another framework like the Operator SDK be sufficient for you? Joining Rook is also
+  about community, not just the framework.
+
+## Engineering Requirements
+
+The engineering costs of each storage provider include:
+
+* Develop the operator
+* Rook maintainers will help answer questions along the way, but ultimately
+  you own the development
+* If there are test failures in the CI, they must be investigated
+* If issues are opened in Github, they need investigation
+* If users have questions in Slack, they need to be answered
```

Comment:
Is use of slack merely recommended or mandatory? Some (of us crazy types) might prefer more FOSS oriented options (IRC/Matix/etc). If it's mandatory it might be helpful to give some guidelines about "they need to be answered" means.

Reply:
I second that, also I wish to centralize communications to Github too but using GitHub discussions they might not be as "live" as Slack but works well with timezones.
However, Slack is THE recommended channel to chat about Rook...

Reply:
Rook community is based around Github and Slack. If a storage provider has other communication channels and directs their users to them, it's fine. But they should also monitor Slack in case questions come up there.